File: GeneratedEntityReaderClasses/Components/PhysicsJointComponent.cs
namespace GeneratedNoitaClasses;
public class PhysicsJointComponent : NoitaComponentBase,  iNoitaType<PhysicsJointComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaBool nail_to_wall { get; set; } = new();
	public NoitaBool grid_joint { get; set; } = new();
	public NoitaBool breakable { get; set; } = new();
	public NoitaUInt type { get; set; } = new();
	public NoitaInt body1_id { get; set; } = new();
	public NoitaInt body2_id { get; set; } = new();
	public NoitaFloat pos_x { get; set; } = new();
	public NoitaFloat pos_y { get; set; } = new();
	public NoitaFloat delta_x { get; set; } = new();
	public NoitaFloat delta_y { get; set; } = new();
	public NoitaBool mMotorEnabled { get; set; } = new();
	public NoitaFloat mMotorSpeed { get; set; } = new();
	public NoitaFloat mMaxMotorTorque { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		nail_to_wall.Read(s);
		grid_joint.Read(s);
		breakable.Read(s);
		type.Read(s);
		body1_id.Read(s);
		body2_id.Read(s);
		pos_x.Read(s);
		pos_y.Read(s);
		delta_x.Read(s);
		delta_y.Read(s);
		mMotorEnabled.Read(s);
		mMotorSpeed.Read(s);
		mMaxMotorTorque.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		nail_to_wall.Write(s);
		grid_joint.Write(s);
		breakable.Write(s);
		type.Write(s);
		body1_id.Write(s);
		body2_id.Write(s);
		pos_x.Write(s);
		pos_y.Write(s);
		delta_x.Write(s);
		delta_y.Write(s);
		mMotorEnabled.Write(s);
		mMotorSpeed.Write(s);
		mMaxMotorTorque.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/PhysicsJointComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaUInt
from BasicNoitaTypes import NoitaInt
from BasicNoitaTypes import NoitaFloat

from NoitaStream import NoitaStream
@dataclass
class PhysicsJointComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    nail_to_wall:NoitaBool = field(default_factory=NoitaBool)
    grid_joint:NoitaBool = field(default_factory=NoitaBool)
    breakable:NoitaBool = field(default_factory=NoitaBool)
    type:NoitaUInt = field(default_factory=NoitaUInt)
    body1_id:NoitaInt = field(default_factory=NoitaInt)
    body2_id:NoitaInt = field(default_factory=NoitaInt)
    pos_x:NoitaFloat = field(default_factory=NoitaFloat)
    pos_y:NoitaFloat = field(default_factory=NoitaFloat)
    delta_x:NoitaFloat = field(default_factory=NoitaFloat)
    delta_y:NoitaFloat = field(default_factory=NoitaFloat)
    mMotorEnabled:NoitaBool = field(default_factory=NoitaBool)
    mMotorSpeed:NoitaFloat = field(default_factory=NoitaFloat)
    mMaxMotorTorque:NoitaFloat = field(default_factory=NoitaFloat)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.nail_to_wall.Read(s)
        self.grid_joint.Read(s)
        self.breakable.Read(s)
        self.type.Read(s)
        self.body1_id.Read(s)
        self.body2_id.Read(s)
        self.pos_x.Read(s)
        self.pos_y.Read(s)
        self.delta_x.Read(s)
        self.delta_y.Read(s)
        self.mMotorEnabled.Read(s)
        self.mMotorSpeed.Read(s)
        self.mMaxMotorTorque.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.nail_to_wall.Write(s)
        self.grid_joint.Write(s)
        self.breakable.Write(s)
        self.type.Write(s)
        self.body1_id.Write(s)
        self.body2_id.Write(s)
        self.pos_x.Write(s)
        self.pos_y.Write(s)
        self.delta_x.Write(s)
        self.delta_y.Write(s)
        self.mMotorEnabled.Write(s)
        self.mMotorSpeed.Write(s)
        self.mMaxMotorTorque.Write(s)

